test/evaluate/ISO-Fortran-binding.cc

```diff
@@ -210,14 +210,459 @@ int main() {
       rank3d, extents);
   MATCH(false,
       dv_3darray->dim[2].extent == 2 + 66);  // extents was read
+}
+
+static void check_CFI_address(
+    const CFI_cdesc_t *dv, const CFI_index_t subscripts[]) {
+  // 18.5.5.2
+  void *addr = CFI_address(dv, subscripts);
+  const Descriptor *desc = reinterpret_cast<const Descriptor *>(dv);
+  void *addrCheck = desc->Element<void>(subscripts);
+  MATCH(true, addr == addrCheck);
+}
+
+// Helper function to set lower bound of descriptor
+static void EstablishLowerBounds(CFI_cdesc_t *dv, CFI_index_t *sub) {
+  for (int i{0}; i < dv->rank; ++i) {
+    dv->dim[i].lower_bound = sub[i];
+  }
+}
+
+// Helper to get size without making internal compiler functions accessible
+static std::size_t ByteSize(CFI_type_t ty, std::size_t size) {
+  CFI_CDESC_T(0) storage;
+  CFI_cdesc_t *dv = reinterpret_cast<CFI_cdesc_t *>(&storage);
+  int retCode =
+      CFI_establish(dv, nullptr, CFI_attribute_other, ty, size, 0, nullptr);
+  return (retCode == CFI_SUCCESS) ? dv->elem_len : 0;
+}
+
+static void run_CFI_address_tests() {
+  // Test CFI_address defined in 18.5.5.2
+  // Create test world
+  CFI_index_t extents[CFI_MAX_RANK];
+  CFI_CDESC_T(CFI_MAX_RANK) dv_storage;
+  CFI_cdesc_t *dv{reinterpret_cast<CFI_cdesc_t *>(&dv_storage)};
+  static char base;
```

Comment:
Why is this static?  Can a stack address not be used?

Reply:
It can, I have removed the static keyword.

Reply:
Okay. This pattern is repeated frequently, so you may want to look for more of these.

Reply:
Thanks, I removed them everywhere but in a function where the address is returned through the arguments. I do not like giving a stack address in this situation even if it is not supposed to be used in this case.

Reply:
If you have to return the address of a static variable, please make sure that it's `const`-qualified.  That should cause the static variable to reside in a read-only page and help detect cases where user code tries to write to it.